### ISSUE TITLE
Optional fields incorrectly appear in Go output

### ISSUE THREAD

**Issue by BlakeOrth** [glideapps/quicktype#2021]
When using a JSON schema to generate output for Go, the `omitempty` tag is not applied to embedded objects that are not required by the schema. This results in the optional fields being rendered as `"key":null` instead of being omitted from the output which causes issues with pedantic JSON parsers.

Example JSON schema:
```json
{
  "id": "http://json-schema.org/geo",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "basic_required": {
        "type": "string"
    },
    "basic_not_required": {
        "type": "string"
    },
    "required": {
        "oneOf": [ { "type": "string" }, { "type": "number" } ]
    },
    "not_required": {
        "oneOf": [ { "type": "string" }, { "type": "number" } ]
    }
  },
  "required": ["basic_required", "required"]
}
```

quicktype generated types:
```go
type Test struct {
	BasicNotRequired *string   `json:"basic_not_required,omitempty"`
	BasicRequired    string    `json:"basic_required"`              
	NotRequired      *Required `json:"not_required"`                
	Required         *Required `json:"required"`                    
}

type Required struct {
	Double *float64
	String *string
}
```

Minimal Reproduction code:
```go
package main

import (
	"encoding/json"
	"fmt"
)

type Test struct {
	BasicNotRequired *string   `json:"basic_not_required,omitempty"`
	BasicRequired    string    `json:"basic_required"`
	NotRequired      *Required `json:"not_required"`
	Required         *Required `json:"required"`
}

type Required struct {
	Double *float64
	String *string
}

func main() {
	t := Test{}
	b, _ := json.MarshalIndent(t, "", "  ")
	fmt.Println(string(b))
}
```

Expected output:
```json
{
  "basic_required": "",
  "required": null
}
```

Actual Output:
```json
{
  "basic_required": "",
  "not_required": null,
  "required": null
}
```

Playground using full code generation and generated `Marshal()`:
https://go.dev/play/p/abGx9rplr03

**Comment by byrnedo** [comment]
Also being bitten by this.
We have a required field that has `type` `string` or `null` in an object in a group of oneOf.
It's missing the `omitempty` tag in the generated struct.

**Comment by inferrinizzard** [comment]
Resolved as of #2449

**Comment by schani** [comment]
Triage note (systematic backlog sweep, assisted by Claude): confirmed reproducible on current master (v23.3.0, `68ecb284`), even though this issue is closed. Noting it because the *default* behavior still matches the original report — only opt-in behavior changed.

**Root cause:** `canOmitEmpty` in `packages/quicktype-core/src/language/Golang/utils.ts` unconditionally excludes the `union`/`null`/`any` kinds from getting an `omitempty` tag, unless the `--omit-empty` option (added in #2449) is passed. With default options, an optional field typed as a union without `null` (e.g. `oneOf: [string, number]`, not required) still generates:
```go
NotRequired *Required `json:"not_required"`
```
with no `omitempty`, so an unset field marshals as `"not_required": null` — which the schema (whose `oneOf` doesn't include `null`) forbids.

**Fix sketch:** #2449 added `--omit-empty` as an opt-in flag rather than fixing the default. A proper default-behavior fix would let `canOmitEmpty` allow `omitempty` for optional unions that don't contain `null`, without needing the flag — a small change in that function, plus updating the Go golden tests.

**Assessment:**
- Severity: medium — wrong default output, but the scenario is narrow (optional non-nullable union fields) and a workaround (`--omit-empty`) exists.
- Difficulty: easy — a few lines in `canOmitEmpty` plus golden-test fixture updates.

**Reproduction:**
```
node quicktype --lang go --src-lang schema --top-level Test test.schema.json
```
```json
{
  "id": "http://json-schema.org/geo",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "basic_required": { "type": "string" },
    "basic_not_required": { "type": "string" },
    "required": { "oneOf": [ { "type": "string" }, { "type": "number" } ] },
    "not_required": { "oneOf": [ { "type": "string" }, { "type": "number" } ] }
  },
  "required": ["basic_required", "required"]
}
```
generates `NotRequired *Required \`json:"not_required"\`` (no `omitempty`) unless `--omit-empty` is passed.
